docs/source/development/style-guide.rst

```diff
@@ -0,0 +1,392 @@
+``imap_processing`` Style Guide
```

Comment:
There are no content changes to this document (except for a few items added in the `Naming Conventions` section), so reviewers don't need to review the content, just the formatting.  (Suggestions for content changes are always welcomed though!)